### PR TITLE
add macOS unittest to GHA

### PR DESCRIPTION
**Pull request by pmeier**
This PR does two things:

1. Per title. In the same vein as we did for the Linux jobs, we will continue to run the macOS jobs on CircleCI for a few weeks to see if everything runs smoothly. This also gives a chance to iron out stuff like #7364.
2. Merge the M1 workflows into the same workflow as the regular x86_64 ones.

cc @seemethere

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -298,6 +298,8 @@ def get_extensions():
     use_jpeg = use_jpeg and jpeg_found
     if use_jpeg:
         print("Building torchvision with JPEG image support")
+        print(f"  libpng include path: {jpeg_include}")
+        print(f"  libpng lib path: {jpeg_lib}")
```
Comment (pmeier) [inline setup.py:302]: Small drive-by that helped me debug. We do the same for PNG already:

https://github.com/pytorch/vision/blob/0e62c34fb6ed7766fea368e67033564d8010ea78/setup.py#L278-L279

I wasn't able to get the version of the library in a way that would hold for all operating systems. I guess knowing the location is good enough.

File: .github/unittest.sh [hunk 2]
```diff
@@ -7,6 +7,30 @@ CONDA_PATH=$(which conda)
...
+
+# The x86 macOS runners, e.g. the GitHub Actions native "macos-12" runner, has some JPEG libraries installed by default
+# that interfere with our build. We uninstall them here and use the one from conda below.
+if [[ "${OS_TYPE}" == "macos" && $(uname -m) == x86_64 ]]; then
+  JPEG_LIBS=$(brew list | grep jpeg)
+  echo $JPEG_LIBS
+  for lib in $JPEG_LIBS; do
+    brew uninstall --ignore-dependencies --force $lib || true
+  done
+fi
```
Comment (pmeier) [inline .github/unittest.sh:32]: We are doing the same for the macOS wheel workflows:

https://github.com/pytorch/vision/blob/0e62c34fb6ed7766fea368e67033564d8010ea78/packaging/pre_build_script.sh#L3-L9

File: .github/unittest.sh [hunk 3]
```diff
@@ -7,6 +7,30 @@ CONDA_PATH=$(which conda)
...
+  Darwin)
+    OS_TYPE=macos
+    ;;
+  *)
+    echo "Unknown OS type:" $(uname)
+    exit 1
+    ;;
+esac
+
+# The x86 macOS runners, e.g. the GitHub Actions native "macos-12" runner, has some JPEG libraries installed by default
```
Comment (osalpekar) [inline .github/unittest.sh:24]: Let's wrap this with the GHA `::group::` so it's clear what's happing in the logs

**Review by osalpekar — APPROVED** [review]
LGTM - thanks!
